src/main/java/com/leverx/learningmanagementsystem/core/security/context/TenantContext.java

```diff
@@ -0,0 +1,28 @@
+package com.leverx.learningmanagementsystem.core.security.context;
+
+public class TenantContext {
+
+    private static final ThreadLocal<String> tenantId = new ThreadLocal<>();
```

Comment:
instead of creation separate ThreadLocal variable for each field, it's better to use Map: `ThreadLocal<Map<String, String>> CONTEXT`